monkestation/code/modules/antagonists/mimic/mimic.dm

```diff
@@ -0,0 +1,398 @@
+	maxHealth = 75
+	health = 75
+	melee_damage = 10 //For some reason it does double damage so leave it at half of what you want it. I'll try to figure out hopefully before this is out.
+	obj_damage = 21 //21 is the minimum damage to hurt doors, which they need since they can't vent crawl.
+	see_in_dark = 8
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	wander = FALSE
+	attacktext = "consumes"
+	attack_sound = 'sound/effects/tableslam.ogg'
+
+	var/playstyle_string = "<span class='big bold'>You are a mimic,</span></b> an alien that made it's way on to the station. \
+							You may take the form of any item nearby by clicking on it. You can latch onto people by clicking on them, \
+							which is instant when you're disguised. When you latch onto someone, they can't hurt you, but other people\
+							can. After someone dies, you can absorb their body and reproduce to make more mimics.</b>"
+
+	var/disguised = FALSE
+	var/atom/movable/form = null
+	var/disguise_time = 0
+	var/people_absorbed = 0
+	//The target npc mimic's try to disguise as.
+	var/atom/movable/ai_disg_target = null
+	//attempts to reach a disguise target
+	var/ai_disg_reach_attempts = 0
+	var/fleeing = FALSE
+	mobchatspan = "blob"
+	discovery_points = 2000
+
+/mob/living/simple_animal/hostile/alien_mimic/examine(mob/user)
+	if(disguised)
+		. = form.examine(user)
+		. += "<span class='warning'>It jitters a little bit...</span>"
+	else
+		. = ..()
+
+/mob/living/simple_animal/hostile/alien_mimic/Destroy()
+	. = ..()
+	ai_disg_target = null
+
+/mob/living/simple_animal/hostile/alien_mimic/med_hud_set_health()
+	if(disguised && !isliving(form))
+		var/image/holder = hud_list[HEALTH_HUD]
+		holder.icon_state = null
+		return //we hide medical hud while disguised
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/med_hud_set_status()
+	if(disguised && !isliving(form))
+		var/image/holder = hud_list[STATUS_HUD]
+		holder.icon_state = null
+		return //we hide medical hud while disguised
+	..()
+
+/mob/living/simple_animal/hostile/alien_mimic/ClickOn(atom/target_item)
+	if(allowed(target_item)) //Become Items
+		if(disguise_time > world.time)
+			to_chat(src, "<span class='warning'>You diguised too recently, wait a little longer!</span>")
+			return
+		if(disguised)
+			restore()
+		var/obj/item/item = target_item
+		if(!item.anchored)
+			disguise(item)
+			return
+	. = ..()
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/allowed(atom/movable/target_item) // make it into property/proc ? not sure if worth it
+	return isitem(target_item) & !istype(target_item, /obj/item/radio/intercom)
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/is_table(atom/possible_table)
+	return istype(possible_table, /obj/structure/table) || istype(possible_table, /obj/structure/rack)
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/should_heal()
+	return health <= MIMIC_HEALTH_FLEE_AMOUNT
+
+/mob/living/simple_animal/hostile/alien_mimic/proc/latch(mob/living/target)
```

Comment:
Add a check and return for non-mob targets.
I just got a runtime on clicking a turf.